04_output.R

```diff
@@ -10,103 +10,60 @@
-
-  }
-}
-
-# Run graphing function
-Mortality(mort_summary)
-
-
-# Save list
+# Write directory for plot outputs
 dir.create("out", showWarnings = FALSE)
-saveRDS(grizz_plotlist, file = "out/grizz_plotlist.rds")
-
-##
-## OPTION TWO --------------------------------------------------------------------
-##
-
-## imap plot list?
-plot_list <- imap(gbpu_names, ~ {
-  print(.y)
-  svgplots <- Mortality(mort_summary)
-  list(plots = svgplots)
-})
 
 # Create list of GBPU
-gbpu_list <- unique(mort_gbpu$gbpu_name)
+gbpu_list <- unique(mort_summary$gbpu_name)
+
+# Create list for plots
+plot_list <- vector(length = length(gbpu_list), mode = "list")
+names(plot_list) <- gbpu_list
+
+# Create plotting function
+Mortality <- function(data, name) {
+  # Create plot for a single GBPU
+  mortality_plot <- ggplot(data, aes(x = hunt_year, y = count,
+                                     fill = kill_code)) +
+    geom_bar(stat = "identity") + # Add bar for each year w/ fill = kill type
+    scale_fill_brewer("Mortality Type", palette = "Set2") +
+    scale_x_continuous(breaks=seq(1970, 2017, by = 5)) +
+    labs(x = "Year", y = "Number of Grizzly Bears Killed",
+         fill = "Mortality Type", caption = caption.text) + # Legend text
+    ggtitle(paste("Mortality History for the '"
+                  , name
+                  , "' Population Unit"
+                  , ", 1976-2017"
+                  ,sep = "")) +
+    theme_soe() + theme(plot.title = element_text(hjust = 0.5), # Centre title
+                        legend.position = "bottom",
+                        plot.caption = element_text(hjust = 0)) # L-align caption
+  mortality_plot
+}
 
-# Barchart function
-charts <- function() {
-  ggplot(mort_gbpu, aes(x = hunt_year, y = count, fill = kill_code)) +
-  geom_bar(stat = "identity") + # Add bar for each year w/ fill = kill type
-  scale_fill_brewer("Mortality Type", palette = "Set2") +
-  scale_x_continuous(breaks=seq(1970, 2017, by = 5)) +
-  labs(x = "Year", y = "Number of Grizzly Bears Killed",
-       fill = "Mortality Type", caption = caption.text) + # Legend text
-  ggtitle(paste("Mortality History for the ' "
-                , "' Population Unit"
-                , ", 1976-2017"
-                ,sep = "")) +
-  theme_bw() + theme(plot.title = element_text(hjust = 0.5), # Centre main title
-                     legend.position = "bottom",
-                     plot.caption = element_text(hjust = 0)) # Left-align caption
+# Create ggplot graph loop
+plots <- for (n in gbpu_list) {
+  print(n)
+  data <- filter(mort_summary, gbpu_name == n)
+  # print(head(data))
+  p <- Mortality(data, n)
+  plot_list[[n]] <- p
+  ggsave(p, file = paste0("out/", n, ".svg"))
 }
 
-# Loop through list
-plot_list <- imap(gbpu_list, ~ {
-  print(.y)
-  mortbarcharts <- charts
-  # Save in list
-  list(barchart = mortbarcharts)
-})
+# Check result
+plot_list[["Valhalla"]]
+
+# Svg function
+save_svg <- function(x, fname, ...) {
```

Comment:
If you reinstall envreportutils (`remotes::install_github("bcgov/envreportutils")`) you will find two new functions `save_svg_px()` and `save_png_retina()` which you can use in place of this.